nupic/research/frameworks/continuous_learning/dendrite_layers.py

```diff
@@ -0,0 +1,174 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from nupic.torch.modules import SparseWeights  # , KWinnersBase, KWinners
+from nupic.torch.modules.k_winners import KWinners2d
+
+
+class DendriteInput(nn.Module):
+    """ Sparse linear layer from previous output to
+    "dendrites" - this is the first part of a module
+    that projects via dendrite segments to output units.
+
+    :param in_dim: input dimension
+
+    :param n_dendrites: total number of dendrites - note this will
+    be an integer multiple of the number of downstream units
+
+    :param threshold: (currently unused) - threshold for an
+    in-development dendritic activation or gating function
+
+    :param weight_sparsity: Weight sparsity of the sparse weights.
+    If weight_sparsity=1, it will default to a standard linear layer.
+    """
+
+    def __init__(self,
+                 in_dim,
+                 n_dendrites,
+                 threshold=2,
+                 weight_sparsity=0.2,
```

Comment:
Note that there will soon be a change to this. It will be backward compatible, but will be good to update this soon.
https://github.com/numenta/nupic.torch/pull/65